### PR TITLE
redshift 1.10 (new formula)

### PR DESCRIPTION
**Pull request by DomT4**
New formula for Redshift.

Closes #35348
Closes https://github.com/Homebrew/homebrew-head-only/pull/81

### CODE REVIEW

**Comment by DomT4** [comment]
Thanks for the release @jonls!

**Comment by DomT4** [comment]
It didn't ignore the URL length here either, and it failed the plist length :confused: 

```
==> brew style redshift
== /usr/local/Library/Formula/redshift.rb ==
C:  3: 86: Line is too long. [86/80]
C: 42:114: Line is too long. [115/80]

1 file inspected, 2 offenses detected
```

**Comment by DomT4** [comment]
I don't think I can ask the bot to check myself, I lack the rights, but this should pass the bot on a re-check now we've done away with the line limits.

File: Library/Formula/redshift.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    </plist>
+    EOS
+  end
+
+  def caveats; <<-EOS.undent
+    A .conf file has not been provided. If you want one, see:
+      http://jonls.dk/redshift/
+    And place it in $HOME/.config
+    EOS
+  end
```
Comment (mikemcquaid) [inline Library/Formula/redshift.rb]: Do we need these caveats?

File: Library/Formula/redshift.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    depends_on "libtool" => :build
+  end
+
+  depends_on "pkg-config" => :build
+  depends_on "intltool" => :build
+  depends_on "gettext"
+  depends_on "glib"
+
+  def install
+    # Despite the naming, 'quartz' here does not = x11 dep.
```
Comment (mikemcquaid) [inline Library/Formula/redshift.rb]: Can remove this comment, Quartz is a (relatively) well known OSX component.

File: Library/Formula/redshift.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  def caveats; <<-EOS.undent
+    A .conf file has not been provided. If you want one, see:
+      http://jonls.dk/redshift/
+    And place it in $HOME/.config
+    EOS
+  end
+
+  test do
+    system "#{bin}/redshift", "-V"
```
Comment (mikemcquaid) [inline Library/Formula/redshift.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/redshift.rb (hunk 2 above)
Comment (DomT4) [inline Library/Formula/redshift.rb]: It was the relatively I was worried about :wink:. Shall remove it though.

File: Library/Formula/redshift.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/redshift.rb]: Kinda. The ` .conf ` files are useful, I actually considered vendoring an example one but wasn't sure how maintainers would feel about it. The location is also important, right now, doesn't accept ` etc ` as the place to put the ` .conf ` file, and I get the feeling people would attempt to place one there and then get confused as to why things weren't changing as desired.

File: Library/Formula/redshift.rb (hunk 3 above)
Comment (DomT4) [inline Library/Formula/redshift.rb]: Not really. I could set ` redshift ` running, but unless you want me to manually explore the process list running and look for ` redshift `, verify it's running fine in that process list, and then kill it after the test, there's not a lot else we can do. Redshift's options are basically ` -V `, ` -h ` or ` redshift `, the latter of which just starts the binary and tints the screen as desired. There's very little in the way of *play with this option* settings. I'll double check for you, but I'm pretty sure that's the case.

File: Library/Formula/redshift.rb (hunk 1 above)
Comment (mikemcquaid) [inline Library/Formula/redshift.rb]: That's fair enough.

File: Library/Formula/redshift.rb (hunk 3 above)
Comment (mikemcquaid) [inline Library/Formula/redshift.rb]: Cool.

File: Library/Formula/redshift.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/redshift.rb]: Just handled locally and pushed.

Comment (DomT4) [inline Library/Formula/redshift.rb]: Cool. Thanks! :+1:

File: Library/Formula/redshift.rb (hunk 1 above)
Comment (DomT4) [inline Library/Formula/redshift.rb]: Reminds me I do need to wander upstream and ask if they'll be kind enough to put in a configure switch for that directory though. I get the feeling using ` $HOME/.config ` is going to be a bit alien to most newer OS X users.

**Pull request closed without merging**